pkg/proxy/epinfo.go

```diff
@@ -28,8 +28,14 @@ var (
 	// DefaultEndpointInfoRegistry is the default instance implementing the
 	// EndpointInfoRegistry interface.
 	DefaultEndpointInfoRegistry logger.EndpointInfoRegistry = &defaultEndpointInfoRegistry{}
+	endpointManager             EndpointLookup
 )
 
+// EndpointLookup is any type which maps from IP to the endpoint owning that IP.
+type EndpointLookup interface {
```

Comment:
the DNSProxy could also use this. `Daemon.lookupEPByIP` (https://github.com/cilium/cilium/blob/master//daemon/fqdn.go#L322:18) is basically wrapping this so we could pass endpointmanager directly into the proxy in `StartDNSProxy` (https://github.com/cilium/cilium/blob/master/daemon/fqdn.go#L237)

Reply:
Created https://github.com/cilium/cilium/issues/9008